src/Microsoft.FeatureManagement/FeatureManager.cs

```diff
@@ -639,5 +620,59 @@ private ContextualFeatureFilterEvaluator GetContextualFeatureFilter(string filte
 
             return filter;
         }
+
+        private async void PublishTelemetry(EvaluationEvent evaluationEvent, CancellationToken cancellationToken)
+        {
+            if (evaluationEvent.FeatureDefinition.TelemetryEnabled)
```

Comment:
If telemetry isn't enabled for the feature, there's no reason to create the evaulation event.